lib/average_end_of_day_balance/main.py

```diff
@@ -41,34 +43,35 @@ def get_average_end_of_day_balance(
     all_pages_utc_endtime = datetime(today.year, today.month, today.day)
     all_pages_starttime = all_pages_utc_endtime - timedelta(days=1e5)
 
-    for institution in institutions_w_depository:
-        institution_id = institution.institution_id
-        institution_depository_balance_records = client.balances.get(
+    inst_coroutines = [
+        api_client.balances.get(
             user_uuid=user_uuid,
-            institution_id=institution_id,
+            institution_id=institution.institution_id,
             utc_starttime=all_pages_starttime,
             utc_endtime=all_pages_utc_endtime,
             account_types=["depository"],
         )
+        for institution in institutions_w_depository
+    ]
+    r = await asyncio.gather(*inst_coroutines)
+    for index, inst_list in enumerate(r):
         depository_balance_records = []
-        for balance_record in institution_depository_balance_records:
-            balance_record_dict = balance_record.dict()
+        for bal_rec in inst_list:
+            balance_rec_dict = bal_rec.dict()
             depository_balance_records.append(
-                {
-                    key: balance_record_dict[key]
-                    for key in ["account_id", "ts", "balance"]
-                }
+                {key: balance_rec_dict[key] for key in ["account_id", "ts", "balance"]}
             )
         institution_balances_df = pd.DataFrame(depository_balance_records)
         institution_balances_df = institution_balances_df.assign(
-            institution_name=institution_id
+            institution_name=institutions_w_depository[index].institution_id
         )
         balance_df_list.append(institution_balances_df)
 
     # 1. Combine all institution balances into single dataframe
-    balances_df = pd.concat(balance_df_list)
-    if balances_df.empty:
-        return None
+    if balance_df_list:
+        balances_df = pd.concat(balance_df_list)
```

Comment:
len(balance_df) can still be zero here, if `balance_df_list` is a list of empty dataframes. we need to check for that.